### ISSUE TITLE
-[RCTModuleMethod initWithObjCMethodName:JSMethodName:moduleClass] doesn't handle non-standard naming

### ISSUE THREAD

**Issue by jwyld** [facebook/react-native#1789]
I'm exporting the following method from a class:
set:(NSString*)key :(id)value

(Yes, that's a valid method signature.)

The problem is that this method distills this down to thinking it's looking for the selector set: on that class.  The attempt to get the NSMethodSignature for this will fail.  From here, everything falls apart as it attempts to

    NSMutableArray *argumentBlocks = [[NSMutableArray alloc] initWithCapacity:numberOfArguments - 2];

That ends up being a capacity of -2.

**Comment by ide** [comment]
As a workaround I believe you could write:
```objc
RCT_REMAP_METHOD(set, setForKey:(NSString *)key value:(id)value)
{
}
```

**Comment by jwyld** [comment]
This is the same issue as Issue #1778.

**Comment by brentvatne** [comment]
@jwyld - any interest in submitting a PR for this?

**Comment by jwyld** [comment]
Sorry, our react-native usage has been put on the back burner for a bit, so I have other pressing matters in the short term.

**Comment by brentvatne** [comment]
@jwyld - no problem. If anyone in the community wants to pick this up, it's up for grabs. Unassigned @nicklockwood as I'm sure he also has some more pressing matters :smile:

**Comment by booxood** [comment]
Has been solved?

**Comment by nicklockwood** [comment]
Yes, this should be fixed now. Went in at the same time as nullability support.

**Comment by booxood** [comment]
@nicklockwood 

Which version has been solved?
I am still have the question at 0.7.1 version.

**Comment by booxood** [comment]
Output:
```
2015-08-11 16:05:23.385 OneIos[960:3083843] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSPlaceholderArray initWithCapacity:]: capacity (18446744073709551614) is ridiculous'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000011024ac65 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x000000010f366bb7 objc_exception_throw + 45
	2   CoreFoundation                      0x00000001101172a5 -[__NSPlaceholderArray initWithCapacity:] + 245
	3   OneIos                              0x000000010ebdb859 -[RCTModuleMethod initWithObjCMethodName:JSMethodName:moduleClass:] + 1961
	4   OneIos                              0x000000010ec1c642 -[RCTModuleData loadMethods] + 450
	5   OneIos                              0x000000010ec1c393 -[RCTModuleData initWithExecutor:uid:instance:] + 387
	6   OneIos                              0x000000010ec10db4 -[RCTBatchedBridge registerModules] + 3892
	7   OneIos                              0x000000010ec0fa61 -[RCTBatchedBridge initWithParentBridge:] + 1633
	8   OneIos                              0x000000010ec34127 -[RCTBridge setUp] + 375
	9   OneIos                              0x000000010ec33809 -[RCTBridge initWithBundleURL:moduleProvider:launchOptions:] + 713
	10  OneIos                              0x000000010ebd4e08 -[RCTRootView initWithBundleURL:moduleName:launchOptions:] + 152
	11  OneIos                              0x000000010eb5b5ed -[AppDelegate application:didFinishLaunchingWithOptions:] + 189
	12  UIKit                               0x0000000111eb8748 -[UIApplication _handleDelegateCallbacksWithOptions:isSuspended:restoreState:] + 240
	13  UIKit                               0x0000000111eb9357 -[UIApplication _callInitializationDelegatesForMainScene:transitionContext:] + 2540
	14  UIKit                               0x0000000111ebc19e -[UIApplication _runWithMainScene:transitionContext:completion:] + 1349
	15  UIKit                               0x0000000111ebb095 -[UIApplication workspaceDidEndTransaction:] + 179
	16  FrontBoardServices                  0x0000000116ae15e5 __31-[FBSSerialQueue performAsync:]_block_invoke_2 + 21
	17  CoreFoundation                      0x000000011017e41c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	18  CoreFoundation                      0x0000000110174165 __CFRunLoopDoBlocks + 341
	19  CoreFoundation                      0x0000000110173f25 __CFRunLoopRun + 2389
	20  CoreFoundation                      0x0000000110173366 CFRunLoopRunSpecific + 470
	21  UIKit                               0x0000000111ebab02 -[UIApplication _run] + 413
	22  UIKit                               0x0000000111ebd8c0 UIApplicationMain + 1282
	23  OneIos                              0x000000010eb678af main + 111
	24  libdyld.dylib                       0x00000001139ad145 start + 1
	25  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by nicklockwood** [comment]
We're up to version 0.9 now.

**Comment by nicklockwood** [comment]
Ah, my mistake - it looks like it wasn't in version 0.9 yet. It's included in the pending release: https://github.com/facebook/react-native/pull/2287

**Comment by booxood** [comment]
:joy: 
Hope  as soon as possible.

**Comment by ide** [comment]
Nick's fix has been published with 0.10.0-rc.
